dlib/dnn/loss.h

```diff
@@ -757,19 +757,19 @@ namespace dlib
                         if(image_rect_to_feat_coord(p, input_tensor, x, x.label, sub, k))
                         {
                             // Ignore boxes that can't be detected by the CNN.
-                            loss -= 1;
+                            loss -= options.loss_per_missed_target;
                             continue;
                         }
                         const size_t idx = (k*output_tensor.nr() + p.y())*output_tensor.nc() + p.x();
                         loss -= out_data[idx];
                         // compute gradient
-                        g[idx] = -scale;
+                        g[idx] = -scale * options.loss_per_missed_target;
```

Comment:
I'm very confident this isn't right.  If you work out the gradient you should find that it's 1.

Reply:
Also, the loss factors do affect the gradient because they affect `truth_score_hits` which determines which parts of the gradient tensor get modified.

Reply:
Anyway, if you look at the MMOD paper and do the math I'm pretty sure the gradient is +/- 1.  That isn't to say that changing the gradient to this is necessarily a bad idea.  It might actually be a better way to tweak the preference for more detections vs. fewer false alarms.  But some experiments are needed to decide if it's actually better.

Reply:
Oh, ok. I admit that I was working mainly based on trial and error. But even if I made `loss_per_missed_target` really high, it didn't seem to affect the results. Guess I just need to try and err more. Thanks for your prompt reply.

Reply:
Right, making it really high doesn't do much.  That's a limitation of how the MMOD loss is formulated.  So honestly, this might be better.  See my other comments about it in this PR :)